polyglot/piranha/src/models/rule.rs

```diff
@@ -231,7 +244,7 @@ impl Rule {
   /// Gets the first match for the rule in `self`
   pub(crate) fn get_edit(
     &self, source_code_unit: &SourceCodeUnit, rule_store: &mut RuleStore, node: Node,
-    recursive: bool,
+    recursive: bool, applied_edit_start: &Option<usize>,
```

Comment:
Ensure that the next reported match occurs before (line-column number wise) the previously reported match.